File: Second Year/Semester II/Web Programming/lab8_aspnet/lab8_aspnet/Models/Room.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace lab8_aspnet.Models
{
    [Table("rooms")]
    public class Room
    {
        [Key]
        [Column("id")]
        public int Id { get; set; }

        [Required]
        [Column("hotel_id")]
        public int HotelId { get; set; }

        [Required]
        [Column("room_number")]
        public string RoomNumber { get; set; } = string.Empty;

        [Required]
        [Column("category")]
        public string Category { get; set; } = string.Empty;

        [Required]
        [Column("type")]
        public string Type { get; set; } = string.Empty;

        [Required]
        [Column("price")]
        public decimal Price { get; set; }

        [ForeignKey("HotelId")]
        public Hotel? Hotel { get; set; }
    }
}

File: Second Year/Semester II/Web Programming/lab9_node/src/models/Room.js
class Room {
    constructor({ id, hotel_id, hotel_name, room_number, category, type, price }) {
        this.id = id;
        this.hotelId = hotel_id;
        this.hotelName = hotel_name;
        this.roomNumber = room_number;
        this.category = category;
        this.type = type;
        this.price = price;
    }
}

module.exports = Room;